src/main/java/com/questhelper/quests/fishingcontest/FishingContest.java

```diff
@@ -0,0 +1,273 @@
+import net.runelite.api.Skill;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.game.FishingSpot;
+
+@QuestDescriptor(
+	quest = QuestHelperQuest.FISHING_CONTEST
+)
+public class FishingContest extends BasicQuestHelper
+{
+	// Required
+	ItemRequirement coins, fishingPass, garlic, fishingRod, spade, redVineWorm, winningFish, trophy;
+
+	// Recommended
+	ItemRequirement combatBracelet, camelotTeleport, food;
+	QuestRequirement fairyRingAccess;
+
+	// Steps
+	QuestStep talkToVestriStep, getGarlic, goToMcGruborWood, goToRedVine, grandpaJack, runToJack, goToHemenster, teleToHemenster;
+	QuestStep putGarlicInPipe, speakToBonzo, fishNearPipes, speakToBonzoWithFish, speaktoVestri;
+
+	ConditionalStep goToHemensterStep, getWorms;
+
+	ItemRequirementCondition hasCombatBracelet, noCombatBracelet, noFishingRod, hasFishingRod, hasGarlic, noGarlic, hasWorms, noWorms, hasFishingPass;
+	Conditions hasEverything, notNearWorms, inWoods, notInWoods, enteredContest, hasPutGarlicInPipe;
+	WidgetTextCondition garlicInPipeScreen, confirmGarlicInPipe;
+
+	// Zones
+	Zone mcGruborWoodEntrance, cmbBraceletTeleportZone, nearRedVineWorms, contestGroundsEntrance;
+	ZoneCondition passedThroughMcGruborEntrance, atCmbBraceletTeleportZone, onContestGrounds;
+
+	public void setupZones()
+	{
+		mcGruborWoodEntrance = new Zone(new WorldPoint(2662, 3500, 0));
+		passedThroughMcGruborEntrance = new ZoneCondition(mcGruborWoodEntrance);
+
+		cmbBraceletTeleportZone = new Zone(new WorldPoint(2651, 3444, 0), new WorldPoint(2657, 3439, 0));
+		atCmbBraceletTeleportZone = new ZoneCondition(cmbBraceletTeleportZone);
+
+		nearRedVineWorms = new Zone(new WorldPoint(2634, 3491, 0), new WorldPoint(2626, 3500, 0));
+
+		contestGroundsEntrance = new Zone(new WorldPoint(2642, 3442, 0), new WorldPoint(2642, 3441, 0));
+		onContestGrounds = new ZoneCondition(contestGroundsEntrance);
+	}
+
+	public void setupItemRequirements()
+	{
+		coins = new ItemRequirement("Coins", ItemID.COINS_995, 5);
+		coins.setTip("10 if you buy a fishing rod from Jack");
+		fishingPass = new ItemRequirement("Fishing Pass", ItemID.FISHING_PASS);
+		fishingPass.setTip("This can be obtained during the quest.");
+		garlic = new ItemRequirement("Garlic", ItemID.GARLIC);
+		garlic.setTip("This can be obtained during the quest.");
+		garlic.setHighlightInInventory(true);
+		fishingRod = new ItemRequirement("Fishing Rod", ItemID.FISHING_ROD);
+		fishingRod.setTip("This can be obtained during the quest.");
+		spade = new ItemRequirement("Spade", ItemID.SPADE);
+		redVineWorm = new ItemRequirement("Red Vine Worm", ItemID.RED_VINE_WORM, 1);
+		food = new ItemRequirement("Food for low levels", -1, -1);
+		food.setDisplayItemId(BankSlotIcons.getFood());
+		winningFish = new ItemRequirement("Raw Giant Carp", ItemID.RAW_GIANT_CARP);
+		winningFish.setHighlightInInventory(true);
+		trophy = new ItemRequirement("Fishing Trophy", ItemID.FISHING_TROPHY);
```

Comment:
A tip of 'If you lost this you can get another from Bonzo in Hemenster' is nice if someone somehow lost it.